libs/hooks/src/mention/useMention.tsx

```diff
@@ -0,0 +1,170 @@
+import { Mentions, MentionProps as AMentionProps, Avatar } from "antd";
+import React, { ReactElement, useEffect, useMemo, useState } from "react";
+import { UserInfo } from "@next-core/brick-types";
+import { makeThrottledAggregation } from "@next-core/brick-utils";
+import {
+  UserAdminApi_searchAllUsersInfo,
+  type UserAdminApi_SearchAllUsersInfoResponseBody,
+} from "@next-sdk/user-service-sdk";
+import { debounce } from "lodash";
+
+const getUserInfoByNameOrInstanceId = makeThrottledAggregation(
+  "getUserInfoByNameOrInstanceId",
+  (ids: string[]) => {
+    return UserAdminApi_searchAllUsersInfo({
+      query: {
+        state: "valid",
+        $or: [
+          {
+            name: {
+              $like: `%${ids}%`,
+            },
+          },
+          {
+            instanceId: {
+              $like: `%${ids}%`,
+            },
+          },
+        ],
+      },
+      fields: {
+        name: true,
+        nickname: true,
+        user_email: true,
+        user_tel: true,
+        user_icon: true,
+        user_memo: true,
+      },
+    });
+  },
+  ({ list }: UserAdminApi_SearchAllUsersInfoResponseBody) => list as UserInfo[]
+);
+
+interface LegacyMentionProps extends AMentionProps {
+  users: UserInfo[];
+  loading: boolean;
+  onSearch?: (text: string, prefix: string) => void;
+}
+
+const LegacyMention: React.FC<LegacyMentionProps> = ({
+  users,
+  loading,
+  onSearch,
+  ...restProps
+}) => {
+  return (
+    <Mentions onSearch={onSearch} loading={loading} {...restProps}>
+      {Array.isArray(users) &&
+        users.map((item) => (
+          <Mentions.Option value={item.name} key={item.name}>
+            <Avatar
+              src={item.user_icon}
+              size={24}
+              style={{
+                backgroundColor: item?.user_icon
+                  ? undefined
+                  : "rgb(0, 113, 235)",
+                marginRight: "10px",
+              }}
+            >
+              {!item.user_icon && item.name?.slice(0, 2)}
+            </Avatar>
+            {item.name}
+          </Mentions.Option>
+        ))}
+    </Mentions>
+  );
+};
+
+type UseMentionConfig = AMentionProps & {
+  debounceTime?: number;
+};
+
+type UseMentionReturnType = {
+  loading: boolean;
+  users: UserInfo[];
+  Mention: ReactElement<AMentionProps, typeof Mentions>;
+  updateConfig: (config: UseMentionConfig) => void;
+  updateUserOrInstanceId: (nameOrInstanceId: string) => void;
+};
+
+/**
+ *
+ * @param nameOrInstanceId
+ * @param config
+ * @return UseMentionReturnType
+ * @example
+ *
+ * function Todo () {
+ *   const {Mention,users,loading,updateConfig,updateUserOrInstanceId} = useMention("fakeUserName", config)
+ *   return <>
+ *     {Mention}
+ *   </>
+ * }
+ */
+export default function useMention(
+  nameOrInstanceId?: string,
```

Comment:
这个参数并不是 `nameOrInstanceId`，不要误导使用者。

```suggestion
  query?: string,
```

Reply:
这里就是`nameOrInstanceId`吧。

Reply:
应该是「根据用户名关键词搜索」吧？